src/Shape/Whisker.js

```diff
@@ -0,0 +1,230 @@
+import {select} from "d3-selection";
+
+import {accessor, assign, BaseClass, configPrep, constant, elem} from "d3plus-common";
+import {nest} from "d3-collection";
+
+import Circle from "./Circle";
+import Line from "./Line";
+import Rect from "./Rect";
+
+const shapes = {Circle, Rect};
+
+/**
+    @class Whisker
+    @extends BaseClass
+    @desc Creates SVG whisker based on an array of data.
+*/
+export default class Whisker extends BaseClass {
+
+  /**
+      @memberof Whisker
+      @desc Invoked when creating a new class instance, and overrides any default parameters inherited from BaseClass.
+      @private
+  */
+  constructor() {
+
+    super();
+
+    this._endpoint = accessor("endpoint", "Rect");
+    this._endpointConfig = {
+      Circle: {
+        r: accessor("r", 5)
+      },
+      Rect: {
+        height: accessor("height", 5),
+        width: accessor("width", 20)
+      }
+    };
+    this._length = accessor("length", 25);
+    this._lineConfig = {};
+    this._orient = accessor("orient", "top");
+    this._x = accessor("x", 0);
+    this._y = accessor("y", 0);
+
+  }
+
+  /**
+      @memberof Whisker
+      @desc Draws the whisker.
+      @param {Function} [*callback*]
+      @chainable
+  */
+  render(callback) {
+
+    if (this._select === void 0) {
+      this.select(select("body").append("svg")
+        .style("width", `${window.innerWidth}px`)
+        .style("height", `${window.innerHeight}px`)
+        .style("display", "block").node());
+    }
+
+    const lineData = [];
+    this._data.forEach((d, i) => {
+
+      const orient = this._orient(d, i);
+
+      let endpointX = this._x(d, i);
+      if (orient === "left") endpointX -= this._length(d, i);
+      else if (orient === "right") endpointX += this._length(d, i);
+
+      let endpointY = this._y(d, i);
+      if (orient === "top") endpointY -= this._length(d, i);
+      else if (orient === "bottom") endpointY += this._length(d, i);
+
+      lineData.push({__d3plus__: true, data: d, i, id: i, x: this._x(d, i), y: this._y(d, i)});
+      lineData.push({__d3plus__: true, data: d, i, id: i, x: endpointX, y: endpointY});
+    });
+
+    const whiskerData = [];
+    this._data.forEach((d, i) => {
```

Comment:
Instead of pushing into `whiskerData` one by one, just use a map:

```js
const whiskerData = this._data.map((d, i) => {
  // logic
  return dataObj;
});
```